### PR TITLE
feat(apig/signature): add new resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new APIG resource support: signature.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new APIG resource support: signature.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccSignature_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccSignature_basic -timeout 360m -parallel 4
=== RUN   TestAccSignature_basic
=== PAUSE TestAccSignature_basic
=== CONT  TestAccSignature_basic
--- PASS: TestAccSignature_basic (479.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      479.227s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_signature.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	error) {
+	importedId := d.Id()
+	parts := strings.SplitN(importedId, "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID, want '<instance_id>/<id>', but got '%s'",
+			importedId)
+	}
+
+	d.SetId(parts[1])
+	return []*schema.ResourceData{d}, d.Set("instance_id", parts[0])
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_signature.go]: although this works, it's recommend

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_signature_test.go [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+  key         = "%[3]s"
+  secret      = "%[4]s"
+}
+
+resource "huaweicloud_apig_signature" "without_key" {
+  instance_id = huaweicloud_apig_instance.test.id
+  name        = "%[2]s_without_key_update"
+  type        = "aes"
+  algorithm   = "aes-256-cfb"
+  key         = "%[3]s%[4]s" # the length of the 256 signature key is 32.
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_signature_test.go]: this line looks quite confusing

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccSignature_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccSignature_ -timeout 360m -parallel 4
=== RUN   TestAccSignature_basic
=== PAUSE TestAccSignature_basic
=== RUN   TestAccSignature_hmac
=== PAUSE TestAccSignature_hmac
=== RUN   TestAccSignature_aes
=== PAUSE TestAccSignature_aes
=== CONT  TestAccSignature_basic
=== CONT  TestAccSignature_aes
=== CONT  TestAccSignature_hmac
--- PASS: TestAccSignature_basic (472.37s)
--- PASS: TestAccSignature_aes (475.33s)
--- PASS: TestAccSignature_hmac (476.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      476.729s
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_signature_test.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_signature_test.go]: fixed.

File: huaweicloud/utils/utils.go [hunk 3]
```diff
@@ -459,3 +459,13 @@ func FlattenResponse(resp *http.Response) (interface{}, error) {
 	}
 	return respBody, nil
 }
+
+// Reverse is a Function that reverses the order of the characters in the
```
Comment (ShiChangkuo) [inline huaweicloud/utils/utils.go]:
```suggestion
// Reverse is a function that reverses the order of the characters in the
```

File: huaweicloud/utils/utils.go [hunk 4]
```diff
@@ -459,3 +459,13 @@ func FlattenResponse(resp *http.Response) (interface{}, error) {
...
+// given string.
+func Reverse(str string) string {
+	result := []byte(str)
+	for i := 0; i < len(str); i++ {
```
Comment (ShiChangkuo) [inline huaweicloud/utils/utils.go]: it's better to use an effective way to reverse a string

**Comment by Lance52259** [comment]
```
go test -v -run=TestAccFunction_reverse
=== RUN   TestAccFunction_reverse
    utils_test.go:69: The processing result of function 'Reverse' meets expectation: "gfedcba"
--- PASS: TestAccFunction_reverse (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils 0.014s
```

File: huaweicloud/utils/utils.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/utils/utils.go]: fixed.

File: huaweicloud/utils/utils.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/utils/utils.go]: The function logic has been optimized.

**Pull request merged**